src/segment_wifi.go

```diff
@@ -0,0 +1,103 @@
+package main
+
+import (
+	"regexp"
+	"strings"
+)
+
+type wifi struct {
+	props              *properties
+	env                environmentInfo
+	State              string
+	SSID               string
+	RadioType          string
+	AuthenticationType string
+	Channel            string
+	ReceiveRate        string
+	TransmitRate       string
+	SignalStrength     string
+}
+
+const (
+	State              = "State"
+	SSID               = "SSID"
+	RadioType          = "Radio type"
+	AuthenticationType = "Authentication"
+	Channel            = "Channel"
+	ReceiveRate        = "Receive rate (Mbps)"
+	TransmitRate       = "Transmit rate (Mbps)"
+	SignalStrength     = "Signal"
+)
+
+func (w *wifi) enabled() bool {
+	// If in Linux, who wnows. Gonna need a physical linux machine to develop this since the wlan interface isn't available in wsl
+	// But possible solution is to cat /proc/net/wireless, and if that file doesn't exist then don't show the segment
+	// In Windows and in wsl you can use the command `netsh.exe wlan show interfaces`. Note the .exe is important for this to worw in wsl.
+	if w.env.getPlatform() == windowsPlatform || w.env.isWsl() {
+		cmd := "netsh"
+		if !w.env.hasCommand(cmd) {
+			return false
+		}
+		cmdResult, err := w.env.runCommand(cmd, "wlan", "show", "interfaces")
```

Comment:
this won't work cross platform so if you want to contribute this, you need to find a way that works everywhere ;-)

Reply:
Sorry, I didn't see these comments. I meant to push the PR to my own repo so I could run the GitHub Actions.

I have a new PR that I just submitted. But it will have this same limitation. Testing this feature is impossible without a physical Linux machine with an actual WiFi network interface. Virtual Linux environments don't have that.

What other things could I do? I noticed the spotify segment has some compilation hints that allows them to have Windows, Linux, and WSL specific implementations. I was thinking of doing something like that, and just excluding Linux/Darwin for now.